tests/BaseElementTest.php

```diff
@@ -236,4 +252,141 @@ public function testOnBeforeWriteNoParent()
 
         $this->assertEquals(0, (int) $element1->Sort);
     }
+
+    public function getElementCMSLinkDataProvider()
+    {
+        return [
+            // Element in DataObject with $directLink === true
+            'element1' => [
+                TestElement::class,
+                'elementDataObject1',
+                'http://localhost/test/1/ItemEditForm/field/ElementalArea/item/',
```

Comment:
This link represents the `CMSEditLink`, correct? Shouldn't it start with `http://localhost/admin/`?

Reply:
I added "test" name for route specially to check that new URL from CMSEditLink method from DataObject. But if you think that "admin" is correct, I'll change.
DONE!